Scripts/Editor/Build/AndroidKeystoreHelper.cs

```diff
@@ -0,0 +1,208 @@
+using Anvil.CSharp.Logging;
+using Anvil.Unity.Editor.UI;
+using UnityEditor;
+using UnityEditor.Build;
+using UnityEditor.Build.Reporting;
+
+namespace Anvil.Unity.Editor.Build
+{
+
+#if UNITY_ANDROID
+    /// <summary>
+    /// Caches keystore information to <see cref="EditorPrefs"/> to that developers don't have to re-enter it each time
+    /// Unity is launched.
+    /// </summary>
+    public class AndroidKeystoreHelper : IPreprocessBuildWithReport
+    {
+        private const string MENU_PATH_PREFIX = "Anvil/Keystore";
+        private static readonly Logger Logger = Log.GetStaticLogger(typeof(AndroidKeystoreHelper));
+
+        private static string PrefsKeyPrefix
+        {
+            get => $"{MENU_PATH_PREFIX}:{PlayerSettings.productName}-{nameof(AndroidKeystoreHelper)}";
+        }
+        private static string KeystorePassPrefsKey
+        {
+            get => $"{PrefsKeyPrefix}-keystorePass";
+        }
+        private static string KeystoreAliasPassPrefsKey
+        {
+            get => $"{PrefsKeyPrefix}-keystoreAliasPass";
+        }
+
+        /// <summary>
+        /// The keystore values that Unity doesn't clear between sessions.
+        /// </summary>
+        private static bool ArePersistentKeystoreValuesSet
+        {
+            get => !string.IsNullOrEmpty(PlayerSettings.Android.keystoreName)
+                   && !string.IsNullOrEmpty(PlayerSettings.Android.keyaliasName);
+        }
+
+        private static bool AreKeystorePasswordsSet
+        {
+            get => !string.IsNullOrEmpty(PlayerSettings.Android.keystorePass)
+                   && !string.IsNullOrEmpty(PlayerSettings.Android.keyaliasPass);
+        }
+
+
+        [MenuItem(MENU_PATH_PREFIX+"/Set Keystore Passwords")]
+        private static void SetKeystorePasswords()
+        {
+            Logger.Debug("Starting...");
+            bool result = ApplyKeystorePasswordsFromUser();
+            Logger.Debug($"Complete. WasSuccessful:{result}");
+        }
+
+        [MenuItem(MENU_PATH_PREFIX+"/Clear Keystore Passwords")]
+        private static void ClearKeystorePasswordsFromPrefs()
```

Comment:
If you wanted to be fancy, there could be a validator for this that disables the menu item when no passwords are currently stored

Reply:
yea, scope creep. Feel free to add if you want. I also considered a menu item with check mark that let you toggle whether this system was enabled at all.